platform/jewel/ui/src/main/kotlin/org/jetbrains/jewel/ui/painter/hints/EmbeddedToInlineCssStyleSvgPatchHint.kt

```diff
@@ -0,0 +1,436 @@
+            if (classAttribute.isBlank()) {
+                element.removeAttribute(classAttributeName)
+                continue
+            }
+
+            // Handle multiple classes: "st0 st9" or single class: "st0"
+            val classNames = classAttribute.trim().split(Regex("\\s+"))
+            val mergedStyleProperties = mergeStylePropertiesFromCssClasses(classNames, cache)
+
+            if (mergedStyleProperties.isEmpty()) {
+                // No styles found, just remove the class attribute
+                element.removeAttribute(classAttributeName)
+                continue
+            }
+
+            // Merge with existing inline styles (inline takes precedence)
+            val existingStyle = element.getAttribute(styleAttributeName)
+            if (existingStyle.isNotEmpty()) {
+                val inlineStyles = parser.parseInlineStyle(existingStyle)
+                // Inline styles override class styles
+                mergedStyleProperties.putAll(inlineStyles)
+            }
+
+            // Set the merged styles
+            val styleValue = mergedStyleProperties.entries.joinToString(";") { "${it.key}:${it.value}" }
+            element.setAttribute(styleAttributeName, styleValue)
+
+            // Remove class attribute after inlining
+            element.removeAttribute(classAttributeName)
+        }
+
+        // Remove all <style> elements from the document
+        removeAllStyleElements()
+    }
+
+    private fun mergeStylePropertiesFromCssClasses(
+        classNames: List<String>,
+        cache: CssClassAttributesCache,
+    ): MutableMap<String, String> {
+        // Resolve and merge styles from all classes
+        val mergedStyles = mutableMapOf<String, String>()
+        for (className in classNames) {
+            val classStyleProperties: Map<String, String>? = cache.getCssPropertiesFor(className)
+            if (classStyleProperties != null) {
+                // Later classes override earlier ones
+                mergedStyles.putAll(classStyleProperties)
+            }
+        }
+        return mergedStyles
+    }
+
+    /** Uses XPath to efficiently query elements with a specific attribute. */
+    private fun Element.getElementsWithAttributeXPath(attributeName: String): List<Element> {
+        val xPath = XPathFactory.newInstance().newXPath()
+
+        val eligibleNodes = xPath.evaluate("//*[@$attributeName]", this, XPathConstants.NODESET) as NodeList
+
+        return buildList {
+            for (i in 0 until eligibleNodes.length) {
+                eligibleNodes.item(i).let { node -> if (node is Element) add(node) }
+            }
+        }
+    }
+
+    /** Removes all <style> elements from the document. */
+    private fun Element.removeAllStyleElements() {
+        val styleElements = getElementsByTagName("style")
+        // Remove in reverse to avoid index shifting issues
+        for (i in styleElements.length - 1 downTo 0) {
```

Comment:
Nit: assuming it exists for this type of collection

```suggestion
        for (i in styleElements.lastIndex downTo 0) {
```